docs/docs/guides/advanced/support_additional_rpc_methods/index.md

```diff
@@ -0,0 +1,114 @@
+---
+sidebar_position: 0
+sidebar_label: Add custom RPC methods
+---
+
+# Add custom RPC methods
+
+#### Introduction
+
+Web3.js is a popular library for interacting with Ethereum blockchain. It provides a set of APIs to interact with Ethereum nodes via JSON-RPC calls. For adding new JSON-RPC function calls to the library, you can do so using the plugin feature in web3.js 4.x. This allows you to extend the functionality of Web3.js and add support for new JSON-RPC methods.
+
+:::caution
+In Web3.js 1.x, `web3.extend()` function could be used to add new JSON-RPC methods. However, this function is not available in Web3.js 4.x. Instead, the plugin feature can be used to achieve the feature of extending web3 functionality.
+:::
+
+Following tutorial will guide you through the process of creating a custom plugin to extend the functionality of web3.js 4.x and add support for new RPC methods.
+
+#### Creating new RPC methods Plugin in 4 Steps
+
+1. First add web3.js as peer dependency in project´s `package.json` and create a TypeScript class for your plugin. This class should extend the `Web3Plugin` class provided by web3.js.
+
+:::info
+This will give your plugin access to [requestManager](/api/web3-core/class/Web3Context#requestManager) and [accountProvider](/api/web3-core/class/Web3Context#accountProvider).
+:::caution
+
+```ts
+import { Web3PluginBase } from 'web3';
+
+export default class CustomRpcMethodsPlugin extends Web3PluginBase {
+	// step 2
+	// ...
+}
+```
+
+2. After that add public `pluginNamespace` property. This will be used to access your plugin, as mentioned in step number 5 code example.
+
+```ts
+import { Web3PluginBase } from 'web3';
+
+export default class CustomRpcMethodsPlugin extends Web3PluginBase {
+	public pluginNamespace = 'customRpcMethods'; // step 3
```

Comment:
I think we do need to have comments like `// step 3` or `// step 4` in documentation especially comment `step 3` in step 2. It is not valuable for users. What do you think @jdevcs?

Reply:
updated